src/ajax/firebase-authentication.html

```diff
@@ -70,51 +70,76 @@
           type: Boolean,
           notify: true
         },
-        _tokens: Object
+        signedIn: Boolean,
+        _tokens: Object,
+        waitingForSignIn: Boolean
       },
+
+      observers: [
+        '_onSignInChange(waitingForSignIn, signedIn)'
+      ],
+
       attached: function() {
         this._parseSuccesfulResult = this._parseSuccesfulResult.bind(this);
       },
+
       /**
-       * Sign in the user with Firebase. It checks if the redirect was succesful.
-       * If that is true, fire `github-token` event with the new token.
-       *
-       * Else it checks for local storage if the token existed and signed in
-       * with the Firebase session and fires `github-token` again.
+       * Perform an auth using a redirect-based sign-in flow.
        */
-      signIn: function(shouldRedirect, privateRepos) {
+      doSignIn: function(privateRepoAccess) {
+        var provider = new firebase.auth.GithubAuthProvider();
+        provider.addScope('user');
+        if (privateRepoAccess) {
+          provider.addScope('repo');
+        } else {
+          provider.addScope('public_repo');
+        }
+        this.$.auth.signInWithRedirect(provider);
+      },
+
+      /**
+       * Check whether the redirect-flow was just finished.
+       * In this case, parse the just obtained GitHub token and continue.
+       * Otherwise, either the user was already signed in, or never signed in.
+       * If the GitHub token is present in local-storage, we wait for firebase
+       * to confirm the signin. If not, we notify that the user was not signed in.
+       */
+      checkSignedIn: function() {
         this.loading = true;
         this.$.auth.auth.getRedirectResult().then(function(result) {
-          if (!result.credential) {
-            if (this.signedIn && this._tokens.GitHub) {
-              this.fire('github-token', this._tokens.GitHub);
-              this.loading = false;
-            } else if (shouldRedirect) {
-              this._redirectToGitHub(privateRepos);
-            }
-          } else {
+          redirectDone = !!result.credential
```

Comment:
This is the first occurrence of redirectDone, why is it not declared?
And the ; is missing.
Just change it to: `var redirectDone = !!result.credential;`